src/euphorie/client/browser/reset_password.py

```diff
@@ -124,6 +124,13 @@ def send_mail(self, email):
             return True
 
         ppr = api.portal.get_tool("portal_password_reset")
+        # Clean out previous requests by this user
+        for token, value in ppr._requests.items():
+            if value[0] == account.id:
+                del ppr._requests[token]
+                ppr._p_changed = 1
+
+        ppr.setExpirationTimeout(0.5)
```

Comment:
You might be interested in this code to clear all the expired stored tokens:
https://github.com/plone/Products.CMFPlone/blob/8c8d4a9a8edb1ba20e1f6987e5880a2f3b55c121/Products/CMFPlone/PasswordResetTool.py#L173-L183

I think that calling `ppr.clearExpired(0.5)` should work (it seems to me it should work with non integer arguments by reading the code).

Reply:
I was kind of expecting that you would point this out. :-) Yes, this is hacky - it does hard code setting the expiration every time a reset is requested. I'll look into improving this. And yes, it's expected to be an integer, but it works as a float (except in the mail template). But let's see if we can get away with one day instead, yes.

`ppr.clearExpired(0.5)` clears out the tokens that have been expired for half a day. The expiration date of a new token will still be in the future, so it will not be cleared until that time.

Reply:
If you can stick with the 12 hours, I would be happy, as it is an explicit requirement and might be part of the retest - which we then would fail.